src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,147 @@
+import { useParams, useSearchParams } from 'react-router-dom';
+import { RenderTable } from '../components/RenderTable';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { ErrorType } from '../types/ErrorType';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[] | null>(null);
+  const [isFetching, setIsFetching] = useState(true);
+  const [errorType, setErrorType] = useState<ErrorType | null>(null);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sort = searchParams.get('sort') || [];
+  const order = searchParams.get('order') || null;
+  const { slug = '' } = useParams();
+
+  const getPeopleFromServer = async () => {
+    try {
+      const fetchedPeople = await getPeople();
+
+      if (!fetchedPeople.length) {
+        setErrorType(ErrorType.NOPEOPLE);
+      }
+
+      setPeople(fetchedPeople);
+    } catch (error) {
+      setErrorType(ErrorType.LOADING);
+    } finally {
+      setIsFetching(false);
+    }
+  };
+
+  useEffect(() => {
+    getPeopleFromServer();
+  }, []);
+
+  const filteredList = useMemo(() => {
+    let filteredPeople = people && [...people];
+
+    if (sex && filteredPeople) {
+      filteredPeople = filteredPeople.filter(person => person.sex === sex);
+    }
+
+    if (query && filteredPeople) {
+      filteredPeople = filteredPeople
+        .filter(person => (
+          person.name.toLowerCase().includes(query)
+          || person.motherName?.toLowerCase().includes(query)
+          || person.fatherName?.toLowerCase().includes(query)
+        ));
+    }
+
+    if (centuries.length && filteredPeople) {
+      filteredPeople = filteredPeople
+        .filter(person => (
+          centuries.includes(Math.ceil(person.died / 100).toString())
+        ));
+    }
+
+    return filteredPeople;
+  }, [sex, query, centuries]);
+
+  useEffect(() => {
+    if (filteredList && !filteredList.length) {
+      setErrorType(ErrorType.SEARCH);
+    } else {
+      setErrorType(null);
+    }
+  }, [filteredList]);
+
+  const sortedList = useMemo(() => (
+    filteredList && filteredList.sort((a, b) => {
+      switch (sort) {
+        case 'name':
+          return order === 'desc'
+            ? b.name.localeCompare(a.name)
+            : a.name.localeCompare(b.name);
+
+        case 'sex':
+          return order === 'desc'
+            ? b.sex.localeCompare(a.sex)
+            : a.sex.localeCompare(b.sex);
+
+        case 'born':
+          return order === 'desc'
+            ? b.born - a.born
+            : a.born - b.born;
+
+        case 'died':
+          return order === 'desc'
+            ? b.died - a.died
+            : a.died - b.died;
+
+        default: break;
```

Comment:
you can make it shorter by using `flag = 1 | -1` to define order

Reply:
fixed it